src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationService.cs

```diff
@@ -51,16 +52,45 @@ public DefaultAuthorizationService(IAuthorizationPolicyProvider policyProvider,
 
             if (authContext.HasSucceeded)
             {
-                _logger.UserAuthorizationSucceeded(user?.Identity?.Name);
+                _logger.UserAuthorizationSucceeded(GetUserNameForLogging(user));
                 return true;
             }
             else
             {
-                _logger.UserAuthorizationFailed(user?.Identity?.Name);
+                _logger.UserAuthorizationFailed(GetUserNameForLogging(user));
                 return false;
             }
         }
 
+        private string GetUserNameForLogging(ClaimsPrincipal user)
+        {
+            var identity = user?.Identity;
+            if (identity != null)
+            {
+                var name = identity.Name;
+                if (name != null)
+                {
+                    return name;
+                }
+                return GetClaimValue(identity, "sub")
+                    ?? GetClaimValue(identity, ClaimTypes.Name)
+                    ?? GetClaimValue(identity, ClaimTypes.NameIdentifier)
+                    ?? GetClaimValue(identity, ClaimTypes.WindowsAccountName)
```

Comment:
As long as it's just for logging (and is not extended to the authentication middleware themselves), I'm fine. 

> Le 31 mars 2016 à 22:04, Hao Kung <notifications@github.com> a écrit :
> 
> In src/Microsoft.AspNetCore.Authorization/DefaultAuthorizationService.cs:
> 
> >                  return false;
> >              }
> >          }
> >  
> > +        private string GetUserNameForLogging(ClaimsPrincipal user)
> > +        {
> > +            var identity = user?.Identity;
> > +            if (identity != null)
> > +            {
> > +                var name = identity.Name;
> > +                if (name != null)
> > +                {
> > +                    return name;
> > +                }
> > +                return GetClaimValue(identity, "sub")
> > +                    ?? GetClaimValue(identity, ClaimTypes.Name)
> > +                    ?? GetClaimValue(identity, ClaimTypes.NameIdentifier)
> > +                    ?? GetClaimValue(identity, ClaimTypes.WindowsAccountName)
> Sure but current behavior is worse as you get user: null which makes it impossible to correlated, at least you have a chance to find the identifier matching a user this way...that's the intent anyways...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>